src/PlusDataCollection/Atracsys/vtkPlusAtracsysTracker.cxx

```diff
@@ -0,0 +1,396 @@
+  {
+    LOG_WARNING(this->Internal->Tracker->ResultToString(result));
+  }
+
+  // make LED blue during pairing
+  this->Internal->Tracker->SetUserLEDState(0, 0, 255, 0);
+
+  // pair active markers
+  if ((result = this->Internal->Tracker->EnableWirelessMarkerPairing(true)) != ATR_SUCCESS)
+  {
+    LOG_ERROR(this->Internal->Tracker->ResultToString(result));
+    return PLUS_FAIL;
+  }
+  LOG_INFO(endl << " *** Put active markers in front of the tracker to pair ***" << endl);
+
+  // sleep while waiting for tracker to pair active markers
+  #ifdef __linux__
+    usleep(1000*this->Internal->ActiveMarkerPairingTimeSec);
+  #endif
+  #ifdef WIN32
+    Sleep(1000*this->Internal->ActiveMarkerPairingTimeSec);
+  #endif
+  
+  LOG_INFO(" *** End of wireless pairing window ***" << endl);
+
+  if ((result = this->Internal->Tracker->EnableWirelessMarkerPairing(false)) != ATR_SUCCESS)
+  {
+    LOG_ERROR(this->Internal->Tracker->ResultToString(result));
+    return PLUS_FAIL;
+  }
+
+  // make LED green, pairing is complete
+  this->Internal->Tracker->SetUserLEDState(0, 255, 0, 0);
+
+  // TODO: check number of active markers paired
+  // TODO: print info about paired markers here
+  
+  cout << "Additional info about paired markers:" << this->Internal->Tracker->GetMarkerInfo() << std::endl;
+
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusAtracsysTracker::InternalDisconnect()
+{
+  LOG_TRACE("vtkPlusAtracsysTracker::InternalDisconnect");
+  this->Internal->Tracker->EnableUserLED(false);
+
+  ATRACSYS_RESULT result;
+  if ((result = this->Internal->Tracker->Disconnect()) != ATR_SUCCESS)
+  {
+    LOG_ERROR(this->Internal->Tracker->ResultToString(result));
+    return PLUS_FAIL;
+  }
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusAtracsysTracker::InternalStartRecording()
+{
+  LOG_TRACE("vtkPlusAtracsysTracker::InternalStartRecording");
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusAtracsysTracker::InternalStopRecording()
+{
+  LOG_TRACE("vtkPlusAtracsysTracker::InternalStopRecording");
+  return PLUS_SUCCESS;
+}
+
+//----------------------------------------------------------------------------
+PlusStatus vtkPlusAtracsysTracker::InternalUpdate()
+{
+  LOG_TRACE("vtkPlusAtracsysTracker::InternalUpdate");
+  const double unfilteredTimestamp = vtkPlusAccurateTimer::GetSystemTime();
+
+  std::vector<AtracsysTracker::Marker> markers;
+
+  ATRACSYS_RESULT result = result = this->Internal->Tracker->GetMarkersInFrame(markers);
+  if (result != ATR_SUCCESS && result != AtracsysTracker::ATRACSYS_RESULT::ERROR_NO_FRAME_AVAILABLE)
+  {
+    LOG_ERROR(this->Internal->Tracker->ResultToString(result));
+    return PLUS_FAIL;
+  }
+  else if (result == AtracsysTracker::ATRACSYS_RESULT::ERROR_NO_FRAME_AVAILABLE)
```

Comment:
move this before the first check and then you can simplify

result != ATR_SUCCESS && result != AtracsysTracker::ATRACSYS_RESULT::ERROR_NO_FRAME_AVAILABLE

to

result != ATR_SUCCESS